cupyx/scipy/special/digamma.py

```diff
@@ -0,0 +1,199 @@
+import cupy
+from cupy import core
+
+
+_digamma_kernel = None
+
+polevl_definition = '''
+static __device__ double polevl(double x, double coef[], int N)
```

Comment:
How about making N a template parameter? I will let the compiler unroll the loop.